src/boot/bootctl-install.c

```diff
@@ -334,6 +337,9 @@ static int update_efi_boot_binaries(const char *esp_path, const char *source_pat
                 _cleanup_close_ int fd = -EBADF;
                 _cleanup_free_ char *v = NULL;
 
+                if (de->d_type != DT_REG)
+                        continue;
+
                 if (!endswith_no_case(de->d_name, ".efi"))
                         continue;
 
```

Comment:
actually, we should be probably more careful with weird inode types here, i.e. at least open the files below with O_NONBLOCK|O_NOCTTY|O_NOFOLLOW in case we hit some weird stuff such as tty device nodes, where we really shouldn't block.